### PR TITLE
Implement V2 Register Page User Feedback

### PR DESCRIPTION
**Pull request by FinnIckler**

### CODE REVIEW

File: app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx [hunk 1]
```diff
@@ -93,11 +96,6 @@ export default function CompetingStep({
           payment: registration.payment,
         },
       );
-      // Going from pending -> Cancelled
-      if (data.registration.competing.registration_status === 'cancelled') {
-        dispatch(setMessage('competitions.registration_v2.register.registration_status.cancelled', 'positive'));
-        return nextStep({ toStart: true });
-      }
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx:100]: Why is this transition/step being dropped?

File: app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx [hunk 2]
```diff
@@ -172,15 +170,21 @@ export default function CompetingStep({
...
+        competition_id: competitionInfo.id,
+        competing: {
+          comment: hasCommentChanged ? comment : undefined,
+          event_ids: hasEventsChanged ? selectedEvents : undefined,
+        },
+        guests,
+      });
+    }).catch(() => {
+      nextStep();
     });
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx:188]: Am I reading it correctly that you're still proceeding to the `nextStep` even if the user does NOT confirm (which is what the `catch` branch represents in this case)?

File: app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx [hunk 3]
```diff
@@ -350,6 +337,9 @@ export default function CompetingStep({
                   {i18n.t('registrations.update')}
                 </Button>
                 <ButtonOr />
+                <Button secondary onClick={() => nextStep()}>
+                  {i18n.t('competitions.registration_v2.register.view_registration')}
+                </Button>
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx:342]: The naming scheme here feels unintuitive to me. The button to **view** your registration (according to the i18n text, at least) internally invokes `nextStep` which in my mind is a mutation. So a button that touts to "view" actually "mutates". Am I missing something here?

File: app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx [hunk 4]
```diff
@@ -37,15 +86,16 @@ export default function RegistrationOverview({
...
         <Header>{i18n.t('competitions.nav.menu.registration')}</Header>
         <Form onSubmit={nextStep} size="large">
           <FormField>
             <label>
               {i18n.t('activerecord.attributes.registration.registration_competition_events')}
               :
             </label>
-            {registration.competing.event_ids.map((id) => (<EventIcon key={id} id={id} style={{ cursor: 'unset' }} />))}
+            { /* Make sure to keep WCA Event order */}
+            {events.official.flatMap((e) => (registration.competing.event_ids.includes(e.id) ? <EventIcon key={e.id} id={e.id} style={{ cursor: 'unset' }} /> : []))}
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx:98]: Maybe do a `filter` first instead of this "hidden filter" by flat-mapping to an empty array?

File: config/locales/en.yml [hunk 5]
```diff
@@ -1775,6 +1775,7 @@ en:
       update:
         being_updated: "Registration is being updated..."
         being_deleted: "Registration is being deleted..."
+        update_confirm: "Are you sure you want to update your Registration?"
```
Comment (gregorbg) [inline config/locales/en.yml:1778]: lower-case `registration`?

File: app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx [hunk 6]
```diff
@@ -63,15 +113,24 @@ export default function RegistrationOverview({
             </label>
             {registration.guests}
           </FormField>
-          { editsAllowed && (
+          <ButtonGroup widths={2}>
```
Comment (gregorbg) [inline app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx:116]: What happens when `deleteAllowed` is false, but the button group width is still set to `2`?

**Review by gregorbg — CHANGES_REQUESTED** [review]

File: app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx (hunk 1 above)
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx:100]: because deletion is now handled on the registration overview

File: app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx (hunk 2 above)
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx:188]: this brings the user back to the RegistrationOverview page

File: app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx (hunk 3 above)
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/CompetingStep.jsx:342]: no mutation happening, the next step from the Update page is the View registration page

File: app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx (hunk 4 above)
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx:98]: I chose to flat map so we only iterate over the array once, sadly JS does not have filterMap

File: app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx (hunk 6 above)
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx:116]: let me see

File: config/locales/en.yml (hunk 5 above)
Comment (FinnIckler) [reply config/locales/en.yml]: fixed

File: app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx (hunk 6 above)
Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx:116]: still works

**Comment by dunkOnIT** [comment]
@thewca-bot deploy staging

File: app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx (hunk 4 above)
Comment (gregorbg) [reply app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx:98]: But you're not iterating over an excessively long array or over something that is costly to compute.

If every array iteration triggered lots and lots of DB calls, then fine. But here, we have a pre-computed list in-memory that is never longer than ~20 entries. Even doing one `includes` call per event is still max 400 calls which literally every processor of any computer less than 20 years old sneezes out in a time faster than I care to find the micro-femto-atto-prefix for :P

TL;DR: Readability and clarity goes over iteration optimization in simple cases like this

Comment (FinnIckler) [reply app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx]: changed!

**Review by gregorbg — APPROVED** [review]
